Tests/BootCamp.Chapter.Tests/DeckTest.cs

```diff
@@ -0,0 +1,50 @@
+﻿using BootCamp.Chapter.Gambling;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace BootCamp.Chapter.Tests
+{
+    public class DeckTest
+    {
+        // I cannot test the happy path on the constructor because the variable that  holds the contents is private
+
+        [Fact]
+        public void New_Given_Null_List_Throws_NullArgumentException()
+        {
+            Assert.Throws<ArgumentNullException>(() => new Deck(null));
+        }
+
+        [Fact]
+        public void DrawFromTop_Given_Empty_List_Throws_OutOfCardException()
+        {
+            var deck = new Deck(new List<Card>() { });
+
+            Assert.Throws<OutOfCardsException>(() => deck.DrawFromTop());
+        }
+
+        [Theory]
+        [MemberData(nameof(deckExpectations))]
+        public void DrawFromTop_Given_List_Return_Last_Item(Deck deck, Card expected)
+        {
+            var actual = deck.DrawFromTop();
+
+            Assert.Equal(expected, actual);
+        }
+
+        // DrawRandom cannot be tested because it needed a mock and we did not learn that on lesson1.
```

Comment:
It can be tested. There are a few cases:
1) DrawRandom, when there is one card- draws that card
2) DrawRandom, when there are no cards- exception

Reply:
written